src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -118,12 +189,205 @@
     return rankings;
   }
 
-  // parse the given file into a List of CastVoteRecords for tabulation
+  // helper to extract the required CVRContest from the input CVR
+  CVRContest getCvrContest(CVR cvr, Contest contestToTabulate) {
+    CVRContest cvrContestToTabulate = null;
+    // find current snapshot
+    CVRSnapshot currentSnapshot = null;
+    if (cvr.CVRSnapshot != null) {
+      for (CVRSnapshot cvrSnapshot : cvr.CVRSnapshot) {
+        if (cvrSnapshot.ObjectId.equals(cvr.CurrentSnapshotId)) {
+          currentSnapshot = cvrSnapshot;
+          break;
+        }
+      }
+    }
+    // find CVRContest which matches the Contest we are tabulating
+    if (currentSnapshot != null && currentSnapshot.CVRContest != null) {
+      for (CVRContest cvrContest : currentSnapshot.CVRContest) {
+        if (cvrContest.ContestId.equals(contestToTabulate.ObjectId)) {
+          cvrContestToTabulate = cvrContest;
+          break;
+        }
+      }
+    }
+    return cvrContestToTabulate;
+  }
+
+  void parseXml(List<CastVoteRecord> castVoteRecords) {
+    try {
```

Comment:
for readability (so you don't have to indent this entire long method), I would move this `try` up a level in the call stack